### PR TITLE
Add delete

### PR DESCRIPTION
**Pull request by mpospelov**

### CODE REVIEW

File: lib/linked-list/list.rb [hunk 1]
```diff
@@ -136,6 +160,11 @@ def each
       __each { |node| yield(node.data) }
     end
 
+    def each_node
```
Comment (spectator) [inline lib/linked-list/list.rb]: Could you please add short documentation for this method? Similar to `each`.

**Review by spectator — CHANGES_REQUESTED** [review]

File: lib/linked-list/list.rb (hunk 1 above)
Comment (mpospelov) [reply lib/linked-list/list.rb]: done 👍

Comment (spectator) [reply lib/linked-list/list.rb]: Thanks! I'll take a closer look tonight and then release a new version.

**Review by spectator — CHANGES_REQUESTED** [review]
@mpospelov looks like there is no test for `each_node`, could you please add it and I think it is good to go after that

**Comment by mpospelov** [comment]
@spectator done 😉

**Review by spectator — APPROVED** [review]

**Comment by spectator** [comment]
thank you @mpospelov ! v0.0.11 is released

**Pull request merged**